plugin.video.svtplay/default.py

```diff
@@ -201,15 +201,30 @@ def __create_dir_item(article, mode):
   if mode == MODE_PROGRAM:
     folder = True
   info = None
-  if "info" in list(article.keys()):
-    if "onlyAvailableInSweden" in article["info"] and \
-        article["info"]["onlyAvailableInSweden"] and \
-        helper.getSetting(S_HIDE_RESTRICTED_TO_SWEDEN):
-      # Do not show geo restricted items
-      return
-    info = article["info"]
+  if __is_geo_restricted(article):
+    logging.log("Hiding geo restricted item {} as setting is on".format(article["title"]))
+    return
+  if not folder and __is_inappropriate_for_children(article):
+    logging.log("Hiding content {} not appropriate for children as setting is on".format(article["title"]))
+    return
+  info = article["info"]
   __add_directory_item(article["title"], params, article["thumbnail"], folder, False, info)
 
+def __is_geo_restricted(program):
+  if program["onlyAvailableInSweden"] and \
```

Comment:
you could have simply done `return program["onlyAvailableInSweden"] and helper.getSettingBool(S_HIDE_RESTRICTED_TO_SWEDEN)`

Reply:
I appreciate the feedback, thanks!